dowhy/causal_estimators/distance_matching_estimator.py

```diff
@@ -103,12 +103,7 @@ def __init__(
         self.matched_indices_att = None
         self.matched_indices_atc = None
 
-    def fit(
-        self,
-        data: pd.DataFrame,
-        exact_match_cols=None,
-        effect_modifier_names: Optional[List[str]] = None,
-    ):
+    def fit(self, data: pd.DataFrame, effect_modifier_names: Optional[List[str]] = None, exact_match_cols=None):
```

Comment:
`exact_match_cols` was ordered incorrectly in arguments that's why the error. Now all function arguments have explicit names.